custom/icds_reports/static/js/directives/awc-reports/awc-reports.directive.js

```diff
@@ -1734,45 +1734,205 @@ function AwcReportsController($scope, $http, $location, $routeParams, $log, DTOp
+                'mData': 'beneficiary',
+                'title': 'Beneficiary Status',
+                'class': 'medium-col',
+                'value': renderBeneficiary
+            },
+            {
+                'mData': 'number_of_thrs_given',
+                'title': 'Number of THRs given in current month',
+                'class': 'medium-col',
+                'value': renderNumberOfThrsGiven
+            },
+            {
+                'mData': 'last_date_thr',
+                'title': 'Date of last THR',
+                'class': 'medium-col',
+                'value': renderLastDateThr
+            },
+        ],
+        'lactating': [
+            {
+                'mData': 'person_name',
+                'title': 'Name',
+                'class': 'big-col',
+                'value': renderPersonNameLactating
+            },
+            {
+                'mData': 'age',
+                'title': 'Age',
+                'class': 'medium-col',
+                'value': renderAge
+            },
+            {
+                'mData': 'add',
+                'title': 'Date of Delivery',
+                'class': 'medium-col',
+                'value': renderAdd
+            },
+            {
+                'mData': 'delivery_nature',
+                'title': 'Type of Delivery',
+                'class': 'medium-col',
+                'value': renderDeliveryNature
+            },
+            {
+                'mData': 'institutional_delivery',
+                'title': 'Institutional Delivery (Y/N)',
+                'class': 'medium-col',
+                'value': renderInstitutionalDeliveryInMonth
+            },
+            {
+                'mData': 'num_pnc_visits',
+                'title': 'Number of PNC visits',
+                'class': 'medium-col',
+                'value': renderNumPncVisits
+            },
+            {
+                'mData': 'breastfed_at_birth',
+                'title': 'Breastfed within hour of birth',
+                'class': 'medium-col',
+                'value': renderBreastfedAtBirth
+            },
+            {
+                'mData': 'is_ebf',
+                'title': 'Exclusively breastfeeding at last home visit',
+                'class': 'medium-col',
+                'value': renderIsEbf
+            },
+            {
+                'mData': 'num_rations_distributed',
+                'title': 'Number of THRs given in current month',
+                'class': 'medium-col',
+                'value': renderNumRationsDistributed
+            }
+        ],
+    };
+
+    vm.dtColumns = [];
+    if (vm.awcReportTableData[vm.step] && !isMobile) {
+        for (var i = 0; i < vm.awcReportTableData[vm.step].length; i++) {
+            var entry = vm.awcReportTableData[vm.step][i];
+            vm.dtColumns.push(DTColumnBuilder.newColumn(entry['mData']).withTitle(entry['title'])
+                .renderWith(entry['value']).withClass(entry['class']));
+        }
```

Comment:
this is great. I didn't review the data structures closely, but assume you were careful when porting them across?

Reply:
yes